openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -603,6 +625,69 @@ Location: https://client.example.net/cb?
   &error_description=Unsupported%20response_type%20value
 ```
 
+# Authorization Challenge Endpoint
+
+An Authorization Server MAY publish parameter `authorization_challenge_endpoint` in its Authorization Server Metadata. In this case, the Wallet SHOULD use this endpoint to obtain authorization. This enables use cases where an Issuer requests Presentation of a Credential before issuing its Credential.
+
+## Authorization Challenge Request
+
+The request to the Authorization Challenge Endpoint follows the same rules as the Pushed Authorization Request (see (#pushed-authorization-request)).
+
+## Authorization Challenge Response
+
+The response from the Authorization Challenge Endpoint can take one of the following forms:
+
+### Error - Request Presentation 
+
+The Authorization Server MAY request a Credential Presentation by responding with the `error` code `insufficient_authorization`. In this case, it MUST include a `presentation` parameter containing an Authorization Request as defined in Section 5 of [@!OpenID4VP], encoded in JSON.
+
+```
+HTTP/1.1 400 Bad Request
+Content-Type: application/json
+Cache-Control: no-store
+
+{
+  "error": "insufficient_authorization",
+  "auth_session": "123456789",
+  "presentation": {
+    "client_id": "x509_san_dns:rp.example.com",
+    "request_uri": "https://rp.example.com/oidc/request/1234"
+  }
+}
+```
+
+If the error code is `insufficient_authorization` and a `presentation` parameter is present, the Wallet MUST process the Authorization Request contained in the `presentation` parameter as defined in [@!OpenID4VP] to perform a Credential Presentation to the Authorization Server. It is then expected to repeat the Authorization Challenge Request.
+
+The Authorization Server MAY use the `auth_session` parameter in the Authorization Challenge Response. The auth session allows the Authorization Server to associate subsequent requests by this Wallet with the ongoing authorization request sequence. The Wallet MUST include the `auth_session` in follow-up requests to the Authorization Challenge Endpoint if it receives one along with the error response.
+
+#todo: Determine whether we needs presentation_during_issuance_session
+
+### Error - Show Website
```

Comment:
Is the show website in this case the "normal" par flow, or just a generic webpage not used for authorization? 

In the first party auth spec the challenge endpoint could fallback to behave like the par endpoint and return a request_uri, i think it will be useful to add this here as well, since a wallet won't know beforehand whether to use authorization challenge or PAR endpoint (so in our wallet we default to authorization challenge endpoint if available).

Currently i see no way to handle the case where normal par must be used and the wallet calls the challenge endpoint

Reply:
I will change the logic to the one defined in the [first-part apps draft](https://www.ietf.org/archive/id/draft-ietf-oauth-first-party-apps-01.html#section-5.2.2.1). This should cover at least use cases 1 and 2 listed by Tobias.